internal/engine/telemetrycontroller.go

```diff
@@ -0,0 +1,82 @@
+package engine
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/windmilleng/wmclient/pkg/dirs"
+
+	"github.com/windmilleng/tilt/internal/build"
+	"github.com/windmilleng/tilt/internal/engine/configs"
+	"github.com/windmilleng/tilt/internal/store"
+	"github.com/windmilleng/tilt/internal/tracer"
+	"github.com/windmilleng/tilt/pkg/model"
+)
+
+type TelemetryController struct {
+	clock build.Clock
+	// lock is a lock that is used to control access from the span exporter and the telemetry controller to the file
+	lock tracer.Locker
+	dir  *dirs.WindmillDir
+}
+
+func NewTelemetryController(lock tracer.Locker, clock build.Clock, dir *dirs.WindmillDir) *TelemetryController {
+	return &TelemetryController{
+		lock:  lock,
+		clock: clock,
+		dir:   dir,
+	}
+}
+
+func (t *TelemetryController) OnChange(ctx context.Context, st store.RStore) {
+	state := st.RLockState()
```

Comment:
can you move this into a function so that other things in this method don't accidentally read `state`? see PodLogManager#diff for an example of this pattern